test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -120,40 +127,74 @@ public String getID() {
         // generate BookkeeperSpec.
         final Map<String, Object> bkPersistentVolumeSpec = getPersistentVolumeClaimSpec("10Gi", "standard");
         // use the latest version of bookkeeper.
-        final Map<String, Object> bookkeeperSpec = ImmutableMap.<String, Object>builder().put("image",
-                                                                                             getImageSpec(DOCKER_REGISTRY + PREFIX + "/" + BOOKKEEPER_IMAGE_NAME, PRAVEGA_BOOKKEEPER_VERSION))
-                                                                                        .put("replicas", bookieCount)
-                                                                                        .put("storage", ImmutableMap.builder()
-                                                                                                                    .put("ledgerVolumeClaimTemplate", bkPersistentVolumeSpec)
-                                                                                                                    .put("journalVolumeClaimTemplate", bkPersistentVolumeSpec)
-                                                                                                                    .build())
-                                                                                        .put("autoRecovery", true)
-                                                                                        .build();
 
         // generate Pravega Spec.
         final Map<String, Object> pravegaPersistentVolumeSpec = getPersistentVolumeClaimSpec("20Gi", "standard");
+        
+        final String bookkeeperImg = DOCKER_REGISTRY + PREFIX + "/" + BOOKKEEPER_IMAGE_NAME;
+        final String pravegaImg = DOCKER_REGISTRY + PREFIX + "/" + PRAVEGA_IMAGE_NAME;
+        final Map<String, Object> bookkeeperImgSpec;
+        final Map<String, Object> pravegaImgSpec;
+
+        if (isLaterOperator) {
+
+            bookkeeperImgSpec = ImmutableMap.of("repository", bookkeeperImg);
+            pravegaImgSpec = ImmutableMap.of("repository", pravegaImg);
+
+        } else {
+
+            bookkeeperImgSpec = getImageSpec(bookkeeperImg, PRAVEGA_BOOKKEEPER_VERSION);
```

Comment:
Do we really need getImageSpec()? we can directly create the ImmutableMap in the else block. That would keep code more readable.

Reply:
Yes as it is being called from other places as well